osquery/config/config.cpp

```diff
@@ -64,7 +64,7 @@ CREATE_REGISTRY(ConfigPlugin, "config");
 CREATE_LAZY_REGISTRY(ConfigParserPlugin, "config_parser");
 
 /// The config plugin must be known before reading options.
-CLI_FLAG(string, config_plugin, "filesystem", "Config plugin name");
+FLAG(string, config_plugin, "filesystem", "Config plugin name");
```

Comment:
As I understand the intent behind `CLI_FLAG` vs `FLAG`, it's that things that change the way osquery works, should be `CLI_FLAG`. I'm leery of changing this one to `FLAG`

Reply:
`config_plugin` cannot be set as a `FLAG` (meaning option) because you need to know how to get the config in order to receive any options.

Sorry for the confusion around this, `CLI_FLAGS` mostly mean, cannot-set-via-config.

Reply:
Is the intent to only have variables relating to config be cli-only (as there's a clear bootstrapping issue)

Or is there an interest in having some class of impactful things require local administer action, and not be freely adjustable. For example, `disable_carver`

Reply:
Originally just the bootstrapping concern.